### PR TITLE
linux: harden terminal parsing

### PR DESCRIPTION
**Pull request by cgzones**
Handle corrupted data in /proc/tty/drivers, either via injection or buffer truncation.

### CODE REVIEW

File: linux/LinuxProcessTable.c [hunk 1]
```diff
@@ -141,20 +141,24 @@ static void LinuxProcessTable_initTtyDrivers(LinuxProcessTable* this) {
    if (r < 0)
       return;
 
-   int numDrivers = 0;
-   int allocd = 10;
+   unsigned numDrivers = 0;
+   unsigned int allocd = 10;
```
Comment (BenBE) [inline linux/LinuxProcessTable.c:145]: Be consistent about the type. Preferably `size_t`.

**Review by BenBE — CHANGES_REQUESTED** [review]
The overall routine could accept some more visual structure as it's currently hard to read in the way it is written. Also, some comment explaining the format to be parsed doesn't hurt.

File: linux/LinuxProcessTable.c [hunk 2]
```diff
@@ -141,20 +141,29 @@ static void LinuxProcessTable_initTtyDrivers(LinuxProcessTable* this) {
...
-   while (*at != '\0') {
+   char* path = NULL;
+   while (at && *at != '\0') {
+      /*
+       * Format:
+       * [name]  [node path]  [major]  [minor range]  [type]
+       * serial  /dev/ttyS    4        64-95          serial
+       */
       at = strchr(at, ' ');    // skip first token
+      if (!at) goto finish;    // bail out on truncation
```
Comment (BenBE) [inline linux/LinuxProcessTable.c:156]: Avoid single-line if statements.

Also maybe leave some space after each block to group the various segment of the parsed line …

File: linux/LinuxProcessTable.c [hunk 3]
```diff
@@ -139,49 +139,73 @@ static void LinuxProcessTable_initTtyDrivers(LinuxProcessTable* this) {
...
+
       token = at;              // mark beginning of minorFrom
       while (*at >= '0' && *at <= '9') at++; //find end of minorFrom
       if (*at == '-') {        // if has range
          *at = '\0'; at++;        // clear and skip
          ttyDrivers[numDrivers].minorFrom = atoi(token); // save
          token = at;              // mark beginning of minorTo
          at = strchr(at, ' ');    // find end of minorTo
+         if (!at)
+                  goto finish;    // bail out on truncation
```
Comment (BenBE) [inline linux/LinuxProcessTable.c:183]: Indentation.

**Review by BenBE — APPROVED** [review]
LGTM. Minor style note.

File: linux/LinuxProcessTable.c (hunk 3 above)
Comment (cgzones) [reply linux/LinuxProcessTable.c]: fixed

**Review by BenBE — APPROVED** [review]

**Pull request merged**